cmd/gb-vendor/vendor/manifest.go

```diff
@@ -89,9 +89,20 @@ type Dependency struct {
 
 // WriteManifest writes a Manifest to the path. If the manifest does
 // not exist, it is created. If it does exist, it will be overwritten.
+// If the manifest file is empty (0 dependencies) it will be deleted.
 // TODO(dfc) write to temporary file and move atomically to avoid
 // destroying a working vendorfile.
 func WriteManifest(path string, m *Manifest) error {
+	if len(m.Dependencies) == 0 {
+		// Check if files exists.
+		_, err := os.Stat(path)
+		if err != nil {
```

Comment:
this might be better written as

err := os.Remove(path) // not removeall, we only want to remove a file, not a directory by mistake
if !os.IsNotExist(err) {
     return err // some other kind of error
}
return nil // not found is ok